File: spring-data-requery-test-java/src/main/java/org/springframework/data/requery/domain/sample/AbstractSite.java
package org.springframework.data.requery.domain.sample;

import io.requery.Entity;
import io.requery.Generated;
import io.requery.Key;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.requery.domain.AbstractPersistable;
import org.springframework.data.requery.domain.ToStringBuilder;

/**
 * AbstractSite
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
@Entity
public abstract class AbstractSite extends AbstractPersistable<Integer> {

    @Key
    @Generated
    protected Integer id;

    protected String name;


    @Override
    protected @NotNull ToStringBuilder buildStringHelper() {
        return super.buildStringHelper()
            .add("name", name);
    }

    private static final long serialVersionUID = -691506293515552942L;
}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/domain/sample/AbstractSite.kt
package org.springframework.data.requery.kotlin.domain.sample

import io.requery.Entity
import io.requery.Generated
import io.requery.Key
import org.springframework.data.requery.kotlin.domain.AbstractPersistable
import org.springframework.data.requery.kotlin.domain.ToStringBuilder
import java.util.*

/**
 * org.springframework.data.requery.kotlin.domain.sample.AbstractSite
 *
 * @author debop
 */
@Entity
abstract class AbstractSite: AbstractPersistable<Int>() {

    @get:Key
    @get:Generated
    abstract override val id: Int

    abstract var name: String


    override fun hashCode(): Int = Objects.hashCode(name)

    @io.requery.Transient
    override fun buildStringHelper(): ToStringBuilder {
        return super.buildStringHelper()
            .add("name", name)
    }
}